### ISSUE TITLE
[Coil] Placeholder plugin doesn't work 

### ISSUE THREAD

**Issue by alaegin** [skydoves/landscapist#243]
**Please complete the following information:**
- Library Version  v2.1.7
- Affected Device(s) Google Pixel 7 Pro with Android 13, Samsung Galaxy S20FE with Android 13
 
**Describe the Bug:**

`PlaceholderPlugin.Loading` doesn't work at all. Even in sample benchmark app.

I've cleaned benchmark activity for testing purposes:

```kotlin
class MainActivity : ComponentActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    setContent {
      val imageComponent = imageComponent {
        +PlaceholderPlugin.Loading(painterResource(id = R.drawable.poster))
      }

      CompositionLocalProvider(LocalImageComponent provides imageComponent) {
        Column {
          CoilImageProfiles()
        }
      }
    }
  }
}

// Image was replaced with extra big one to ensure the loading delay.
@Composable
fun CoilImageProfiles() {
  CoilImage(
    modifier = Modifier.size(120.dp),
    imageModel = { "https://www.kenrockwell.com/nikon/d810/sample-images/file-types/810_3435-LARGE-FINE-OPT-QUAL.JPG" },
    component = LocalImageComponent.current,
  )
}
```

There should be exactly one placeholder and exactly one image then. But, when I open the app I see only a white screen and the target image after several seconds.

I've tried to debug this issue and found out that Loading state was not called at all inside the `onImageStateChanged` (Seems like that placeholder works based on these states)

The entire state log:

```
2023-03-14 20:59:06.986  3854-3854  System.out  com...ves.benchmark.landscapist.app  I  com.skydoves.landscapist.coil.CoilImageState$None@de0d496
2023-03-14 20:59:07.073  3854-3854  System.out  com...ves.benchmark.landscapist.app  I  com.skydoves.landscapist.coil.CoilImageState$None@de0d496
2023-03-14 20:59:10.697  3854-3854  System.out  com...ves.benchmark.landscapist.app  I  Success(drawable=android.graphics.drawable.BitmapDrawable@8d542e2, dataSource=NETWORK)
```
It had just been showing the None state for several seconds until the image was loaded.

Reproduction:

```kotlin
@Composable
fun CoilImageProfiles() {
  CoilImage(
    modifier = Modifier.size(120.dp),
    imageModel = { "https://www.kenrockwell.com/nikon/d810/sample-images/file-types/810_3435-LARGE-FINE-OPT-QUAL.JPG" },
    component = LocalImageComponent.current,
    onImageStateChanged = {
      println(it) // <- Prints only None and Success states
    }
  )
}
```

**Expected Behavior:**

I expect placeholder plugin to call placeholder image while target image is being loaded.

**Comment by alaegin** [comment]
Video of this issue:
https://user-images.githubusercontent.com/10329086/225110857-98b8707a-daa1-4e2c-9f38-bbafe72d682c.mp4

**Comment by skydoves** [comment]
This issue was solved in the new release, `2.1.8`. Thank you for reporting this important issue!
